scripts/lib/CIME/utils.py

```diff
@@ -777,6 +777,78 @@ def compute_total_time(job_cost_map, proc_pool):
 
     return current_time
 
+def format_time(time_format, input_format, input_time):
+    """
```

Comment:
This was needed because CIME was failing with allocation walltimes greater than 23:59:59. This was a limitation of the Python date parser; it doesn't accept >=24 in the hours field; instead expecting it to roll over in to the days field.